WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/feedback/ipp/GetIPPFeedbackBannerData.kt

```diff
@@ -39,58 +43,52 @@ class GetIPPFeedbackBannerData @Inject constructor(
-        if (numberOfTransactionsInLast30Days == null || numberOfTransactionsInLast30Days < 0) {
-            logger.e(AppLog.T.API, "Transactions count data is malformed.")
-            return null
-        }
+        val wcPayIPPOrdersInLast30Days = getIppOrdersforLast(STATS_TIME_WINDOW_LENGTH_DAYS)
+        val hasIPPOrdersInLast30Days = wcPayIPPOrdersInLast30Days.isNotEmpty()
+        val ippOrdersInLast30Days = wcPayIPPOrdersInLast30Days.size
 
-        return when (numberOfTransactionsInLast30Days) {
-            0 -> if (hasUserEverMadeIppTransaction()) {
-                IPP_BEGINNER_BANNER
-            } else {
+        return when {
+            !hasWCPayIPPOrders && cashOnDeliverySettings.isCashOnDeliveryEnabled() -> {
                 IPP_NEWBIE_BANNER
             }
-            in IPP_BEGINNER_TRANSACTIONS_RANGE -> IPP_BEGINNER_BANNER
-            else -> IPP_NINJA_BANNER
+            hasIPPOrdersInLast30Days && ippOrdersInLast30Days in IPP_BEGINNER_TRANSACTIONS_RANGE -> {
+                IPP_BEGINNER_BANNER
+            }
+            wcPayIPPOrdersCount >= IPP_NINJA_TRANSACTIONS_THRESHOLD -> {
+                IPP_NINJA_BANNER
+            }
+            else -> {
+                null
+            }
         }
     }
 
-    private fun WooPayload<WCPaymentTransactionsSummaryResult>.isSuccessful(): Boolean {
-        return !isError && result != null
+    private suspend fun getIppOrders(): List<OrderEntity> {
+        return orderStore.getOrdersForSite(siteModel).filter { isIPPOrders(it) }
     }
 
-    @Suppress("ReturnCount")
-    private suspend fun hasUserEverMadeIppTransaction(): Boolean {
-        val activePaymentsPlugin = checkNotNull(getActivePaymentsPlugin())
-
-        if (activePaymentsPlugin != WCInPersonPaymentsStore.InPersonPaymentsPluginType.WOOCOMMERCE_PAYMENTS) {
-            return false
-        }
-
-        val response = ippStore.fetchTransactionsSummary(activePaymentsPlugin, siteModel)
-
-        if (!response.isSuccessful()) {
-            logger.e(AppLog.T.API, "Error fetching transactions summary: ${response.error.message}")
-            return false
+    private suspend fun getIppOrdersforLast(days: Int): List<OrderEntity> {
+        val timeWindowStartDate = Date().daysAgo(days)
+        return orderStore.getOrdersForSite(siteModel).filter { orderEntity ->
+            isIPPOrders(orderEntity) && isIPPOrderDateAfter(orderEntity, timeWindowStartDate)
         }
+    }
 
-        val numberOfTransactions = response.result?.transactionsCount ?: return false
+    private fun isIPPOrderDateAfter(
+        it: OrderEntity,
+        timeWindowStartDate: Date?
+    ) = SimpleDateFormat(
+        "yyyy-MM-dd",
+        Locale.getDefault()
+    ).parse(it.datePaid)?.after(timeWindowStartDate) == true
 
-        return numberOfTransactions > 0
-    }
+    private fun isIPPOrders(it: OrderEntity) = it.getMetaDataList().any { wcMetaData ->
```

Comment:
```suggestion
    private fun isIPPOrder(order: OrderEntity) = it.getMetaDataList().any { wcMetaData ->
```
🔍 If I understand correctly, this is just a typo. Could be also changed to extension (fun or prop):
```kotlin
private val OrderEntity.isIPP: Boolean
    get() = getMetaDataList().any { wcMetaData ->
        wcMetaData.key == "receipt_url"
    } && paymentMethod == WOOCOMMERCE_PAYMENTS_PAYMENT_TYPE
```

Reply:
Definitely a good suggestion. Thank you. I'll create a separate PR for this. I need to modify the fluxc code as well for this change.

Reply:
Let me clarify this a bit more @AnirudhBhat. I mean that the function name could be `isIPPOrder` instead of `isIPPOrders`. 

As for the idea to make it an extension instead, I don't think any changes are required in FluxC. We could simply declare the above private extension property in the `GetIPPFeedbackBannerData` class.